datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
I did not realize you were still relying on the new ``get_or_create`` method. I was expecting your final commit/flush to potentially generate an exception.

Personally, I find this to be more difficult to reason about and I would do it all one way or the other (I favor the ORM), but as I said before I'm not responsible for maintaining this. As long as those who do have that responsibility are satisfied, I'm satisfied.